### DISCUSSION TITLE
[Feature] API Access

### DISCUSSION THREAD

**Discussion by Gorcenski** [johanohly/AirTrail#69] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

This is a lovely and rapidly changing tool. One thing that would be nice is if there was a programmatic API, especially given that OAuth2 is implemented. I've sadly been unsuccessful at trying to login programmatically, perhaps due to my own ignorance, but in my ideal use case I'd be able to integrate this with my other travel/flight tracking tools and connect them all via programmatic APIs. If this is already possible, some documentation would be great.

**Comment by johanohly** [comment]
Thank you for the feedback! 
I’m currently on holiday for about a week, but the idea of exposing some functionality through an API is definitely interesting. I’ll investigate how to implement this once I’m back. 
In the meantime, could you share which features you'd most like to access via the API? That would help me prioritize the investigation.

**Reply by Gorcenski** [reply]
Sure thing. The #1 feature that I would add would be the Add Flight API. My goal is to be able to add flights to the tracker, as well as my CalDAV server, among other things, automatically by forwarding a booking email. I fly often (50+ flights/year) so it would be great to be able to have a script that automates things.

**Reply by johanohly** [reply]
The next release will, at the minimum, have flight list, create, update and delete (preview docs: https://airtrail.johan.ohly.dk/docs/api/flight/save-flight). Sorry for not coming back to you sooner!

**Comment by johanohly** [comment]
This has been added in v2.0.0!

**Comment by Gorcenski** [comment]
Just saw this—this is great, thanks so much!
